### PR TITLE
Add cache for KaldiReader

### PR DESCRIPTION
**Pull request by david20181**
Make KaldiReader faster by using cache of readers

### CODE REVIEW

File: lhotse/features/io.py [hunk 1]
```diff
@@ -942,6 +942,26 @@ def write(self, key: str, value: np.ndarray) -> str:
...
 """
 
+def check_kaldi_native_io_installed():
+    if not is_module_available("kaldi_native_io"):
+        raise ValueError(
+            "To read Kaldi feats.scp, please 'pip install kaldi_native_io' first."
+        )
+
+
+@lru_cache(maxsize=None)
```
Comment (pzelasko) [inline lhotse/features/io.py:952]: Can you also update

https://github.com/lhotse-speech/lhotse/blob/faddd86d43f247f227ce581a7630c4ce74084c3e/lhotse/features/io.py#L389

to clear the objects cached by this function?

Also maybe I'm wrong but I think we once supported caching for kaldi_io/kaldi_native_io but for very large Kaldi data dirs it took a lot of memory.

Comment (david20181) [reply lhotse/features/io.py:956]: Good suggestion on clearing cache. I updated accordingly.

I've tested on some fairly large data dirs and have not observed memory issue.

Comment (pzelasko) [reply lhotse/features/io.py:956]: Thanks for confirming that! LGTM, please just fix the formatting.

**Comment by pzelasko** [comment]
Thanks!

**Pull request merged**